modules/core/src/test/java/org/apache/ignite/internal/GridDiscoveryManagerChangeCoordinatorTest.java

```diff
@@ -67,23 +66,35 @@
      */
     @Test
     public void testChangeCoordinatorLogging() throws Exception {
-        //Start 2 server nodes
+        //Start two server and one client nodes
         IgniteEx srv1 = startGrid("server1");
-        IgniteEx client = startGrid("client");
-        startGrid("server2");
+        startGrid("client");
+        IgniteEx srv2 = startGrid("server2");
+
+        String srv2Id = srv2.cluster().localNode().id().toString();
 
-        srv1.cluster().active();
+        srv1.cluster().state(ACTIVE);
+
+        LogListener lsnr = LogListener.matches(CRD_CHANGE_MSG)
+                .andMatches(srv2Id)
+                .build();
+
+        listeningLog.registerListener(lsnr);
 
         stopGrid("server1");
         startGrid("server1");
 
-        LogListener lsnr = LogListener.matches(CRD_CHANGE_MSG).build();
+        assertTrue(lsnr.check());
+
+        lsnr.reset();
+
+        LogListener lsnr2 = LogListener.matches(CRD_CHANGE_MSG).build();
 
         listeningLog.registerListener(lsnr);
 
         stopGrid("client");
 
-        assertFalse(lsnr.check());
+        assertFalse(lsnr2.check());
```

Comment:
lsnr2 isn't registered anywhere, do you really need it? 
You can reuse lsnr, as it was reset.